### DISCUSSION TITLE
Possible bug in Sceneform?

### DISCUSSION THREAD

**Discussion by RGregat** [sceneview/sceneform-android#100] [Q&A]
I found this Ticket https://github.com/google-ar/arcore-android-sdk/issues/1170 on the ARCore Repo. This person mentioned a bug inside of `GLHelper.java::createCameraTexture:56`. Sceneform is always looking for the first camera texture id. But it is possible to set a different camera texture id during the session configuration phase.

```java
// Create a camera config filter for the session.
CameraConfigFilter filter = new CameraConfigFilter(session);

// Return only camera configs that target 30 fps camera capture frame rate.
filter.setTargetFps(EnumSet.of(CameraConfig.TargetFps.TARGET_FPS_30));

// Return only camera configs that will not use the depth sensor.
filter.setDepthSensorUsage(EnumSet.of(CameraConfig.DepthSensorUsage.DO_NOT_USE));

// Get list of configs that match filter settings.
// In this case, this list is guaranteed to contain at least one element,
// because both TargetFps.TARGET_FPS_30 and DepthSensorUsage.DO_NOT_USE
// are supported on all ARCore supported devices.
List<CameraConfig> cameraConfigList = session.getSupportedCameraConfigs(filter);

// Use element 0 from the list of returned camera configs. This is because
// it contains the camera config that best matches the specified filter
// settings.
session.setCameraConfig(cameraConfigList.get(0)); // or .get(1), .get(2) depends on your phone.
```

I'm not sure if this is really a bug, because I don't see any regular crashes. I have only from time to time a random crash, but i have no idea if the crash is related to the camera texture thingy.
I put now the above code to choose always the first camera texture id in my configuration phase. If if helps? I'm not sure, but I give it a try. 
What do you guys think about that?

**Comment by RGregat** [comment]
Maybe there is no connection between the camera id and the used camera texture id.. 

But I guess it becomes an issue if you choose during the session configuration a different CameraConfig than camera id 0

**Comment by ThomasGorisse** [comment]
Hi @RGregat,

In fact, from what I think and saw, there is no link between the `cameraTextureId` and the physical camera id.

Sceneform creates an openGL texture and them tell ARCore to draw the camera image (chosen by ARCore from the camera config) on it.
Then the `CameraStream` applies the texture to a renderable added to the Filament Scene.
That's how the magical works.

> I found this Ticket google-ar/arcore-android-sdk#1170 on the ARCore Repo. This person mentioned a bug inside of GLHelper.java::createCameraTexture:56. Sceneform is always looking for the first camera texture id. But it is possible to set a different camera texture id during the session configuration phase.

Sorry for him but this person is totally wrong when he says:

> However, in the legacy code of sceneform which I am still using, it is by default using the first texture it can find, which belongs to camera id 0.

His issue can appear maybe if the `session.setCameraConfig()` is called too lately.

> I put now the above code to choose always the first camera texture id in my configuration phase. If if helps? I'm not sure, but I give it a try.

Is this part really useful? Do you see a difference on the actual used camera?
I was hopping that the default (the best/more accurate) camera was chosen by ARCore when there is no `session.setCameraConfig()` call.

**Reply by RGregat** [reply]
I guess the default one picked by ARCore ist the best one.
